shiny/ui/fill/_fill.py

```diff
@@ -0,0 +1,266 @@
+    "is_fill_carrier",
+    "is_fill_item",
+    "is_fillable_container",
+)
+
+TagT = TypeVar("TagT", bound="Tag")
+
+
+FILL_ITEM_CLASS = "html-fill-item"
+FILL_CONTAINER_CLASS = "html-fill-container"
+
+
+@add_example()
+def as_fillable_container(
+    tag: TagT,
+) -> TagT:
+    tag_prepend_class(tag, FILL_CONTAINER_CLASS)
+    tag.append(fill_dependency())
+    return tag
+
+
+@add_example()
+def as_fill_item(
+    tag: TagT,
+) -> TagT:
+    """
+    Coerce a tag to a fill item
+
+    Filling layouts are built on the foundation of _fillable containers_ and _fill
+    items_ (_fill carriers_ are both _fillable containers_ and _fill items_). This is
+    why most UI components (e.g., :func:`~shiny.ui.card`,
+    :func:`~shiny.ui.layout_sidebar`) possess both `fillable` and `fill` arguments (to
+    control their fill behavior). However, sometimes it's useful to add, remove, and/or
+    test fillable/fill properties on arbitrary :class:`~htmltools.Tag`, which these
+    functions are designed to do.
+
+    Parameters
+    ----------
+    tag
+        a Tag object.
+
+    Returns
+    -------
+    :
+        The original :class:`~htmltools.Tag` object (`tag`) with additional attributes
+        (and an :class:`~htmltools.HTMLDependency`).
+
+    See Also
+    --------
+    * :func:`~shiny.ui.fill.as_fill_carrier`
+    * :func:`~shiny.ui.fill.as_fillable_container`
+    * :func:`~shiny.ui.fill.remove_all_fill`
+    * :func:`~shiny.ui.fill.is_fill_carrier`
+    * :func:`~shiny.ui.fill.is_fill_item`
+    * :func:`~shiny.ui.fill.is_fillable_container`
+    """
+    tag_prepend_class(tag, FILL_ITEM_CLASS)
+    tag.append(fill_dependency())
+    return tag
+
+
+@add_example()
+def as_fill_carrier(
+    tag: TagT,
+) -> TagT:
+    """
+    Make a tag a fill carrier
+
+    Filling layouts are built on the foundation of _fillable containers_ and _fill
+    items_ (_fill carriers_ are both _fillable containers_ and _fill items_). This is
+    why most UI components (e.g., :func:`~shiny.ui.card`,
+    :func:`~shiny.ui.layout_sidebar`) possess both `fillable` and `fill` arguments (to
+    control their fill behavior). However, sometimes it's useful to add, remove, and/or
+    test fillable/fill properties on arbitrary :class:`~htmltools.Tag`, which these
+    functions are designed to do.
+
+    Parameters
+    ----------
+    tag
+        a Tag object.
+
+    Returns
+    -------
+    :
+        The original :class:`~htmltools.Tag` object (`tag`) with additional attributes
+        (and an :class:`~htmltools.HTMLDependency`).
+
+    See Also
+    --------
+    * :func:`~shiny.ui.fill.as_fill_item`
+    * :func:`~shiny.ui.fill.as_fillable_container`
+    * :func:`~shiny.ui.fill.remove_all_fill`
+    * :func:`~shiny.ui.fill.is_fill_carrier`
+    * :func:`~shiny.ui.fill.is_fill_item`
+    * :func:`~shiny.ui.fill.is_fillable_container`
+    """
+    as_fillable_container(tag)
+    as_fill_item(tag)
+    return tag
+
+
+def remove_all_fill(
```

Comment:
Do we use this? Same for `is_fill` methods.

If no, maybe we should not export them.